File: 09_ukazatele/01_11_return_pointer.c
#include <stdio.h>

int* get_pointer(void) {
    // lokální proměnná
    int i = 42;

    // vracíme adresu lokální proměnné
    return &i;
}

int main(void) {
    // získání adresy lokální proměnné
    int *p = get_pointer();

    // tisk adresy
    printf("%p\n", p);

    // přístup na adresu
    printf("%d\n", *p);

    return 0;
}

File: 09_ukazatele/01_11_return_pointer.go
// Ukazatele v jazyku Go
//
// - funkce vracející ukazatel na lokální proměnnou

package main

import "fmt"

func getPointer() *int {
	// lokální proměnná
	var i int = 42

	// vracíme adresu lokální proměnné
	return &i
}

func main() {
	// získání adresy lokální proměnné
	p := getPointer()

	// přístup na adresu
	fmt.Printf("%v\n", p)

	// přístup na adresu
	fmt.Printf("%d\n", *p)
}
